libs/parallelism/algorithms/include/hpx/parallel/datapar/iterator_helpers.hpp

```diff
@@ -490,6 +528,44 @@ namespace hpx { namespace parallel { namespace util { namespace detail {
             }
         }
     };
+
+    struct datapar_transform_loop_step_ind
+    {
+        template <typename F, typename InIter, typename OutIter>
+        HPX_HOST_DEVICE HPX_FORCEINLINE static void call1(
+            F&& f, InIter& it, OutIter& dest)
+        {
+            typedef
+                typename std::iterator_traits<InIter>::value_type value_type;
+
+            typedef typename traits::vector_pack_type<value_type, 1>::type V1;
+
+            invoke_vectorized_inout1_ind<V1>::call_aligned(
+                std::forward<F>(f), it, dest);
+        }
+
+        ///////////////////////////////////////////////////////////////////
+        template <typename F, typename InIter, typename OutIter>
+        HPX_HOST_DEVICE HPX_FORCEINLINE static void callv(
+            F&& f, InIter& it, OutIter& dest)
+        {
+            typedef
+                typename std::iterator_traits<InIter>::value_type value_type;
+
+            typedef typename traits::vector_pack_type<value_type>::type V;
+
+            if (!is_data_aligned(it) || !is_data_aligned(dest))
+            {
+                invoke_vectorized_inout1_ind<V>::call_unaligned(
+                    std::forward<F>(f), it, dest);
+            }
+            else
+            {
+                invoke_vectorized_inout1_ind<V>::call_aligned(
+                    std::forward<F>(f), it, dest);
+            }
```

Comment:
Would it be ok to assume that `callv()` will be called only if `it` and `dest` are aligned? Would an `HPX_ASSERT` be sufficient (instead of having a conditional expression)?

Reply:
@hkaiser yes, that's perfectly alright, as `callv()` is only used only for vector calls since unaligned calls are redirected to `call1()`.

Reply:
Somehow this crept back into the code. I thought you had already changed this function to use an assert.